core/loadpoint.go

```diff
@@ -595,8 +599,19 @@ func (lp *LoadPoint) publishSoC() {
 	if lp.SoC.AlwaysUpdate || lp.connected() {
 		f, err := lp.vehicle.ChargeState()
 		if err == nil {
-			lp.socCharge = f
-			lp.log.DEBUG.Printf("vehicle soc: %.0f%%", lp.socCharge)
+
+			if lp.SoC.EstimateSoC {
+				/* new soc value read from car api */
+				if f != lp.socChargeCar {
+					lp.socChargeCar = f
```

Comment:
Hier jetzt davor
`m = (f - lp.socChargeCar) / (lp.chargedEnergy - lp.chargedEnergySoc)`
und man hat die Energiemenge pro SOC-Prozentpunkt (Steigung) und mit
`capacitySoc = m * 100` hast du dann schon direkt die geschätze Speicherkapazität des Akkus die statt dem nachfolgenden Wert aus der Konfig-Datei benutzt werden kann.

Reply:
Genau im Updatezeitppunkt hast du dann eine Div/0. Ich würde hier auf dem Wert aus der Konfig bleiben, unterliegt weniger Schwnakungen :-)
Was wäre der Vorteill die Kapazität zu berechnen wenn sie eh in der Konfig ist?

Reply:
Der Vorteil ist, dass die Kapazität in der Konfig quasi nie stimmt, da die echte nutzbare Kapazität für den Nutzer unbekannt ist (ungleich Prospektangabe), erheblichen Schwankungen unterliegt (z. B. durch Zelltemperaturen) und auch der Energieverbrauch der sonstigen Fahrzeugsysteme beim Laden sonst nicht in die Berechnung mit einfließt. Insbesondere bei niedrigen Leistungen und langen Ladezeiten kann dieser Anteil ganz erheblich sein.

Reply:
Genau im Updatezeitpunkt ist die Energiedifferenz = 0.
Mir ist es eher egal ob man den Konfigwert verwendet oder den berechneten. Es muss halt konsisten sein und dann überall der Selbe verwendet werden. Zur Einführung der neuen Funktion würde ich aber sagen: keep it simple.

Reply:
`capacitySoc = m * 100` hätte dann die Einheit 1 / Wh. In der Division gehört Nenner und Zähler getauscht.

Reply:
Sehr richtig.